src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseTests.cs

```diff
@@ -2554,28 +2554,26 @@ public async Task AppAbortIsLogged()
         }
 
         [Fact]
-        [Repeat(100)]
         [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/24164")]
         public async Task AppAbortViaIConnectionLifetimeFeatureIsLogged()
         {
             var testContext = new TestServiceContext(LoggerFactory);
 
-            await using (var server = new TestServer(async httpContext =>
+            var closeTask = Task.CompletedTask;
```

Comment:
Might be better to use a TCS so there's no risk of middleware completing before its thread "sees" the closeTask get reassigned. Volatile should also work, but a TCS seems more straight foward.

Reply:
Sure it's unlikely, but I don't see why that wouldn't be possible. `closeTask` isn't volatile. If we're trying to deflake a test, I don't want to take any chances.

Using a TCS guarantees that you have the memory barriers through its internal use of volatile and Interlocked.